examples/routing_service/shmem_udp_gateway/qos.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0"?>
+
+<!-- (c) Copyright, Real-Time Innovations, 2024.  All rights reserved.
+RTI grants Licensee a license to use, modify, compile, and create derivative
+works of the software solely for use with RTI Connext DDS. Licensee may
+redistribute copies of the software provided that all such copies are subject
+to this license. The software is provided "as is", with no warranty of any
+type, including any warranty for fitness for any purpose. RTI is under no
+obligation to maintain or support the software. RTI shall not be liable for
+any incidental or consequential damages arising out of the use or inability
+to use the software. -->
+
+<dds xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://community.rti.com/schema/7.2.0/rti_dds_qos_profiles.xsd">
+    <qos_library name="example_library">
+
+        <qos_profile name="shmem_profile" base_name="BuiltinQosLib::Generic.Common">
+            <domain_participant_qos>
+                <discovery>
+                    <initial_peers>
+                        <element>$(APPS)@builtin.shmem://</element>
```

Comment:
You should probably set a default for each environment variable as follows:

```xml
<configuration_variables>
    <value>
        <element>
            <name>APPS</name>
            <value>5</value>
        </element>
        <element>
            <name>CDS_IP_ADDRESS</name>
            <value>127.0.0.1</value>
        </element>
        <element>
            <name>CDS_PORT</name>
            <value>8000</value>
        </element>
    </value>
</configuration_variables>
```

Reply:
@fgaranda I wouldn't be comfortable suggesting this to a customer. This can lead to inconsistencies if a developer "blindly" trusts these default variables without setting them outside. It is a general services recommendation to have the application complain and fail if variables are not set. Using the <configration_variables> tag goes against that recommendation, since it'll allow the application to just work and not complain